core/injector/application_injector.cpp

```diff
@@ -449,13 +449,21 @@ namespace {
             typename Injector>
   auto choose_runtime_implementation(
       const Injector &injector,
-      application::AppConfiguration::RuntimeExecutionMethod method) {
+      application::AppConfiguration::RuntimeExecutionMethod method,
+      application::AppConfiguration::RuntimeInterpreter interpreter) {
     using RuntimeExecutionMethod =
         application::AppConfiguration::RuntimeExecutionMethod;
+    using RuntimeInterpreter =
+        application::AppConfiguration::RuntimeInterpreter;
     switch (method) {
       case RuntimeExecutionMethod::Interpret:
-        return std::static_pointer_cast<CommonType>(
-            injector.template create<sptr<InterpretedType>>());
+        if (interpreter == RuntimeInterpreter::Binaryen) {
+          return std::static_pointer_cast<CommonType>(
+              injector.template create<sptr<InterpretedType>>());
+        } else if (interpreter == RuntimeInterpreter::WasmEdge) {
+          return std::static_pointer_cast<CommonType>(
+              injector.template create<sptr<CompiledType>>());
+        }
```

Comment:
May add error for `interpreter == Off`

Reply:
I would reconsider these CLI params after we've decided whether we keep Binaryen or not.